moveit_ros/move_group/src/default_capabilities/save_geometry_to_file_service_capability.cpp

```diff
@@ -0,0 +1,84 @@
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2024, PickNik Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Bilal Gill */
+
+#include "save_geometry_to_file_service_capability.h"
+
+#include <moveit/moveit_cpp/moveit_cpp.h>
+#include <moveit/move_group/capability_names.h>
+#include <moveit/utils/logger.hpp>
+
+#include <fstream>
+
+namespace move_group
+{
+
+namespace
+{
+rclcpp::Logger getLogger()
+{
+  return moveit::getLogger("save_geometry_to_file_service");
+}
+}  // namespace
+
+SaveGeometryToFileService::SaveGeometryToFileService() : MoveGroupCapability("save_geometry_to_file")
+{
+}
+
+void SaveGeometryToFileService::initialize()
+{
+  save_geometry_service_ = context_->moveit_cpp_->getNode()->create_service<moveit_msgs::srv::SaveGeometryToFile>(
+      SAVE_GEOMETRY_TO_FILE_SERVICE_NAME,
+      [this](const std::shared_ptr<moveit_msgs::srv::SaveGeometryToFile::Request>& req,
+             const std::shared_ptr<moveit_msgs::srv::SaveGeometryToFile::Response>& res) {
+        res->success = false;
+        std::ofstream file(req->file_path_and_name);
+        if (!file.is_open())
+        {
+          res->success = false;
+          return;
+        }
+        planning_scene_monitor::LockedPlanningSceneRO locked_ps(context_->planning_scene_monitor_);
+        locked_ps->saveGeometryToStream(file);
```

Comment:
Dump question: Have you checked that these file operation don't throw? (This is just a nit question and not supposed to block the PR)

Reply:
I have not, I can look into it more.